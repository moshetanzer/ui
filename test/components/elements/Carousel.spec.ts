import { describe, it, expect } from 'vitest'
import { UCarousel } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Carousel', () => {
  it.each([
    ['with arrows and indicators', { props: { arrows: true, indicators: true } }],
    ['with arrows only', { props: { arrows: true, indicators: false } }],
    ['with indicators only', { props: { arrows: false, indicators: true } }],
    ['with custom previous button', { props: { arrows: true } }],
    ['with custom next button', { props: { nextButton: { label: 'Custom next button', arrows: true } } }],
    ['with prev slot', { slots: { prev: () => 'prev slot' }, props: { arrows: true } }],
    ['with next slot', { slots: { next: () => 'next slot' }, props: { arrows: true } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UCarousel.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UCarousel)
    expect(html).toMatchSnapshot()
  })
})
