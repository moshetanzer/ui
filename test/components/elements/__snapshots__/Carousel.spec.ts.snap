// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel > renders with arrows and indicators correctly 1`] = `
"<div data-v-2a90c2d4="" class="relative">
  <div data-v-2a90c2d4="" class="relative w-full flex overflow-x-auto snap-x snap-mandatory scroll-smooth no-scrollbar"></div>
  <div data-v-2a90c2d4="" class="flex items-center justify-between"><button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:first-child]:rotate-180 absolute left-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Prev"><span class="i-heroicons-chevron-left-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button><button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:last-child]:rotate-180 absolute right-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Next"><span class="i-heroicons-chevron-right-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button></div>
  <div data-v-2a90c2d4="" class="absolute flex items-center justify-center gap-3 bottom-4 inset-x-0"></div>
</div>"
`;

exports[`Carousel > renders with arrows only correctly 1`] = `
"<div data-v-2a90c2d4="" class="relative">
  <div data-v-2a90c2d4="" class="relative w-full flex overflow-x-auto snap-x snap-mandatory scroll-smooth no-scrollbar"></div>
  <div data-v-2a90c2d4="" class="flex items-center justify-between"><button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:first-child]:rotate-180 absolute left-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Prev"><span class="i-heroicons-chevron-left-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button><button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:last-child]:rotate-180 absolute right-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Next"><span class="i-heroicons-chevron-right-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Carousel > renders with custom next button correctly 1`] = `
"<div data-v-2a90c2d4="" class="relative">
  <div data-v-2a90c2d4="" class="relative w-full flex overflow-x-auto snap-x snap-mandatory scroll-smooth no-scrollbar"></div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`Carousel > renders with custom previous button correctly 1`] = `
"<div data-v-2a90c2d4="" class="relative">
  <div data-v-2a90c2d4="" class="relative w-full flex overflow-x-auto snap-x snap-mandatory scroll-smooth no-scrollbar"></div>
  <div data-v-2a90c2d4="" class="flex items-center justify-between"><button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:first-child]:rotate-180 absolute left-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Prev"><span class="i-heroicons-chevron-left-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button><button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:last-child]:rotate-180 absolute right-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Next"><span class="i-heroicons-chevron-right-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`Carousel > renders with indicators only correctly 1`] = `
"<div data-v-2a90c2d4="" class="relative">
  <div data-v-2a90c2d4="" class="relative w-full flex overflow-x-auto snap-x snap-mandatory scroll-smooth no-scrollbar"></div>
  <!--v-if-->
  <div data-v-2a90c2d4="" class="absolute flex items-center justify-center gap-3 bottom-4 inset-x-0"></div>
</div>"
`;

exports[`Carousel > renders with next slot correctly 1`] = `
"<div data-v-2a90c2d4="" class="relative">
  <div data-v-2a90c2d4="" class="relative w-full flex overflow-x-auto snap-x snap-mandatory scroll-smooth no-scrollbar"></div>
  <div data-v-2a90c2d4="" class="flex items-center justify-between"><button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:first-child]:rotate-180 absolute left-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Prev"><span class="i-heroicons-chevron-left-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button>next slot</div>
  <!--v-if-->
</div>"
`;

exports[`Carousel > renders with prev slot correctly 1`] = `
"<div data-v-2a90c2d4="" class="relative">
  <div data-v-2a90c2d4="" class="relative w-full flex overflow-x-auto snap-x snap-mandatory scroll-smooth no-scrollbar"></div>
  <div data-v-2a90c2d4="" class="flex items-center justify-between">prev slot<button data-v-2a90c2d4="" type="button" disabled="" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium text-sm gap-x-1.5 p-1.5 shadow-sm text-white dark:text-gray-900 bg-gray-900 hover:bg-gray-800 disabled:bg-gray-900 dark:bg-white dark:hover:bg-gray-100 dark:disabled:bg-white focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center rtl:[&_span:last-child]:rotate-180 absolute right-4 top-1/2 transform -translate-y-1/2 rounded-full" aria-label="Next"><span class="i-heroicons-chevron-right-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button></div>
  <!--v-if-->
</div>"
`;
